internal/bminventory/inventory_test.go

```diff
@@ -10414,17 +10413,18 @@ func createInventory(db *gorm.DB, cfg Config) *bareMetalInventory {
 	mockCRDUtils = NewMockCRDUtils(ctrl)
 	mockOperatorManager = operators.NewMockAPI(ctrl)
 	mockIgnitionBuilder = ignition.NewMockIgnitionBuilder(ctrl)
-	mockProviderRegistry := registry.NewMockProviderRegistry(ctrl)
 	mockInstallConfigBuilder = installcfg.NewMockInstallConfigBuilder(ctrl)
 	mockHwValidator = hardware.NewMockValidator(ctrl)
 	mockStaticNetworkConfig = staticnetworkconfig.NewMockStaticNetworkConfig(ctrl)
 	dnsApi := dns.NewDNSHandler(cfg.BaseDNSDomains, common.GetTestLog())
 	gcConfig := garbagecollector.Config{DeregisterInactiveAfter: 20 * 24 * time.Hour}
+	pr := registry.InitProviderRegistry(common.GetTestLog())
```

Comment:
The way we test `bminventory` is using mocks for UT. When we want to test the integration between the components we do it in the subsystem tests.